src/sparseml/exporters/transforms/matmul_to_matmulinteger_add_cast_mul.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from onnx import ModelProto, TensorProto
+
+from sparseml.exporters.transforms import OnnxTransform
+from sparseml.exporters.transforms.utils import (
+    INITIALIZER_MATCH,
+    MatchResult,
+    add_quantized_conv_matmul_add_ops,
+    delete_quant_node,
+    get_quantization_params,
+    get_structural_matches,
+    optional_node,
+)
+from sparseml.onnx.utils import (
+    ONNXGraph,
+    get_init_by_name,
+    remove_node_and_params_from_graph,
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+__all__ = ["MatMulToMatMulIntegerAddCastMul"]
+
+
+class MatMulToMatMulIntegerAddCastMul(OnnxTransform):
+    """
+    A transform for converting a MatMul with kernel and bias into a
+    quantized representation
+
+    ```
+    | Starting with:
+    |                        weight (initializer)
+    |                            |
+    |                      QuantizeLinear
+    |                            |
+    |          INPUT     DequantizeLinear
+    |            |               |
+    |     DequantizeLinear   Transpose
```

Comment:
noting this transpose should definitely be optional (looking at onnx modifier support)